### PR TITLE
Adding more test cases before removing RegEx for parameter matching

### PR DESCRIPTION
**Pull request by kulmugdha**
@rynowak -- These are just the new test cases that I have added. Will work on the dev code soon.

### CODE REVIEW

File: test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs [hunk 1]
```diff
@@ -68,6 +83,33 @@ public void ParseRouteParameter_ConstraintWithArgumentsAndOptional_ParsedCorrect
 
             Assert.Single(templatePart.InlineConstraints);
             Assert.Single(templatePart.InlineConstraints, c => c.Constraint == @"test(\d+)");
+
+            templatePart = ParseParameter(@"param:test(\d+)=abc?");
```
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs]: **never** do this kind of thing. Convert it to data-driven test (`[Theory]`).

Also the second kind of case you added here is a different scenario right? One of these has a default value and one doesn't.

File: test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs [hunk 2]
```diff
@@ -57,6 +57,21 @@ public void ParseRouteParameter_ConstraintAndOptional_ParsedCorrectly()
         }
 
         [Fact]
+        public void ParseRouteParameter_ConstraintAndOptional_WithDefaultValue_ParsedCorrectly()
+        {
+            // Arrange & Act
+            var templatePart = ParseParameter(@"param:int=12?");
```
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs]: What happens if we double up the question mark? What about one of the other delimiters.

File: test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs (hunk 1 above)
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs]: Ah, I see this is the same code as the test below, yeah just remove this.

File: test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs [hunk 3]
```diff
@@ -86,6 +128,38 @@ public void ParseRouteParameter_ChainedConstraints_ParsedCorrectly()
         }
 
         [Fact]
+        public void ParseRouteParameter_ChainedConstraints_ColonInPattern_ParsedCorrectly()
+        {
+            // Arrange & Act
+            var templatePart = ParseParameter(@"param:test(\d+):test(\w:+)");
```
Comment (rynowak) [inline test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs]: Also interesting to look at doubling up the delimiters here

**Comment by rynowak** [comment]
Would like to see a few tests for doubling up of delimiters. Other than that looks like a good set.

**Comment by rynowak** [comment]
:watch:

File: test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs (hunk 1 above)
Comment (kulmugdha) [inline test/Microsoft.AspNet.Routing.Tests/InlineRouteParameterParserTests.cs]: yeah sorry forgot to remove this.

**Comment by kulmugdha** [comment]
@rynowak added a few more tests. Please have a look.

**Comment by rynowak** [comment]
:shipit:

**Comment by Eilon** [comment]
Already merged: https://github.com/aspnet/Routing/commit/ddfe5e9736aed167aa785563c7fc7b203372eb50

**Pull request closed without merging**